### PR TITLE
fix: Set X-Requested-With header on all requests to avoid browser auth dialogs

### PR DESCRIPTION
**Pull request by juliusknorr**
This used to work for most cases in the far past as jquery adds this on requests (https://github.com/jquery/jquery/blob/48cc402a917d6011c7d3e75f779f11ef91b474fb/src/ajax/xhr.js#L43-L50)

https://github.com/nextcloud-libraries/nextcloud-axios/pull/637 adds this to our axios wrapper but this PR handles cases where apps or code parts are not using that.

Server has logic in place to return a dummyauth response header in case this one is set:
https://github.com/nextcloud/server/commit/dfc3536d2b95fea1b54b3e85651e3a66c2d0088e

This helps to handle requests that happen in the web UI after authentication was revoked (e.g. by disabling a user) more gracefully, otherwise a request would lead to the browser showing a basic auth dialog.

Steps to reproduce:
- Login as a user
- Disable the user through occ
- Wait until the notifications app sends another request

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by artonge — APPROVED** [review]

**Comment by artonge** [comment]
In which cases do we not use axios?

**Comment by juliushaertl** [comment]
Old code or apps that might not use our libraries, rare cases but might still happen

**Review by skjnldsv — APPROVED** [review]

**Review by Pytal — APPROVED** [review]

File: core/src/utils/xhr-request.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+			}
+			if (!init.headers) {
+				init.headers = new Headers()
+			}
+
+			if (init.headers instanceof Headers && !init.headers.has('X-Requested-With')) {
+				init.headers.append('X-Requested-With', 'XMLHttpRequest')
+			} else if (init.headers instanceof Object && !init.headers['X-Requested-With']) {
+				init.headers['X-Requested-With'] = 'XMLHttpRequest'
+			}
```
Comment (skjnldsv) [inline core/src/utils/xhr-request.js:50]: @juliushaertl a small find as of today:
image

If `headers` is an instance of `Headers` and already have `X-Requested-With`, then the second `else if` will still be true.
Luckily, `['X-Requested-With']` on a `Headers` object is ignored by fetch :see_no_evil:

**Pull request merged**